include/kernels/madwf_tensor.cuh

```diff
@@ -68,6 +68,9 @@ namespace quda
       constexpr Tensor5DReduce(const Arg &arg) : arg(arg) { }
       static constexpr const char *filename() { return KERNEL_FILE; }
 
+      // overload comm_reduce to prevent any global reduction
+      static inline void comm_reduce(std::vector<reduce_t> &) { }
```

Comment:
This approach seems prone to "gotchas". Can we do something along the lines of a `push`/`pop` disabling global reductions in stretches of code instead? (Maybe we already have something like this?)

Reply:
push/pop is what we want to be avoiding here.  A kernel functor should know what type of reduction it's doing, and if it's doing a global or local reduction.

Reply:
I'm reminded that this type of mechanism does exist and can be queried by `commGlobalReduction();`

Reply:
This abstraction also allows us, for example, to implement custom global reductions for a given functor.  E.g., a time-slice reduction for a contraction could be implemented here (or we could define it in a parent class) and derive from it.

Reply:
Yes, we have the push/pop functionality.  That's more for the case where we want to enable/disable global reductions at run time, e.g., switching between a Schwarz preconditioner and the outer solver.  This functionality here is for hard disablement at compile time, where know we never want to have a global reduction, and / or provide a custom one.

Reply:
Okay, I now understand and appreciate the compile time vs runtime motivation, thank you for explaining that. I think the way it's implemented is still a bit prone to confusion---overriding a global-scope function with a `static inline` class routine, if I'm gathering it correctly.

While it's not something I'm worried about here (we're in a *.cuh), I could fathom a situation where:
 * I override `comm_reduce` in my class as such
 * Within my class, I call some function that includes reductions, but is defined in another `*.o` in a context where `comm_reduce` has _not_ been overriden
 
In this contrived example, I think I'd get "unexpected" (though strictly correct) behavior, where some functions have global reductions enabled, while others don't.

This might be too contrived to be worth worrying about in practice, but I think it's technically still a risk?

Reply:
Not following the "global-scope" aspect of this.  It's a static class function, and not global.

Reply:
Resolved offline---I missed (aka overlooked reading) how `comm_reduce` is a member of the parent reduction class, so I ended up running myself into a bit of a confused circle.